### PR TITLE
Added hierarchy acceptance tests

### PR DESCRIPTION
**Pull request by Matt-Guest**
What

New acceptance tests for the hierarchy api

How to review

review the code and run the tests agains the api

Who can review

anyone

### CODE REVIEW

File: hierarchyAPI/get_node_hierarchy_test.go [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+		err := datastore.TeardownHierarchy()
+		if err != nil {
+			log.ErrorC("Unable to tear down test data", err, nil)
+			os.Exit(1)
+		}
+	})
+}
+
+func TestErrorStatesGetNodeHierarchy(t *testing.T) {
```
Comment (nshumoogum) [inline hierarchyAPI/get_node_hierarchy_test.go:67]: I know you have followed the spec, but if an instance or dimension against an instance is not found is this not a bad request when trying to find hierarchy node?

Either way there are two tests missing, one to return error when `instance_id` is wrong (not found -> resulting in a bad request) and `dimension_name` is wrong (not found -> resulting in a bad request).

File: hierarchyAPI/get_root_hierarchy_test.go [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+		err := datastore.TeardownHierarchy()
+		if err != nil {
+			log.ErrorC("Unable to tear down test data", err, nil)
+			os.Exit(1)
+		}
+	})
+}
+
+func TestErrorStatesGetRootHierarchy(t *testing.T) {
```
Comment (nshumoogum) [inline hierarchyAPI/get_root_hierarchy_test.go:67]: Again a test missing to check error is returned when the instance_id exists but the dimension name is incorrect.

And once again i think the spec is wrong, if instance doesn't exist then return status 400 bad request, if dimension name not found then return status 404 not found.

**Review by nshumoogum — CHANGES_REQUESTED** [review]

File: hierarchyAPI/get_node_hierarchy_test.go (hunk 1 above)
Comment (Matt-Guest) [reply hierarchyAPI/get_node_hierarchy_test.go:67]: I added the additional tests, you have requested.

File: hierarchyAPI/get_node_hierarchy_test.go [hunk 3]
```diff
@@ -71,21 +71,35 @@ func TestErrorStatesGetNodeHierarchy(t *testing.T) {
...
-			}
 
+		err := datastore.Setup()
+		if err != nil {
+			log.ErrorC("Unable to setup test data", err, nil)
+			os.Exit(1)
+		}
+		// This should return 400 but returns 404
+		SkipConvey("When a child hierarchy node is requested with a invalid instance", func() {
+			Convey("Then a 404 response code is returned", func() {
```
Comment (nshumoogum) [inline hierarchyAPI/get_node_hierarchy_test.go]: Typo `404` should be `400`

File: hierarchyAPI/get_node_hierarchy_test.go [hunk 4]
```diff
@@ -71,21 +71,35 @@ func TestErrorStatesGetNodeHierarchy(t *testing.T) {
...
+				response := hierarchyAPI.GET("/hierarchies/{instance_id}/{dimension_name}/{code}", "0000", "aggregate", "cpi1dim1G120100")
+				response.Expect().Status(http.StatusBadRequest)
+			})
+		})
+		// This should return 400 but returns 404
+		SkipConvey("When a child hierarchy node is requested with a invalid dimension name", func() {
 			Convey("Then a 404 response code is returned", func() {
```
Comment (nshumoogum) [inline hierarchyAPI/get_node_hierarchy_test.go]: As above, should be `400` in test title

File: hierarchyAPI/get_root_hierarchy_test.go [hunk 5]
```diff
@@ -69,23 +69,29 @@ func TestErrorStatesGetRootHierarchy(t *testing.T) {
...
-			}
+		err := datastore.Setup()
+		if err != nil {
+			log.ErrorC("Unable to setup test data", err, nil)
+			os.Exit(1)
+		}
+		// This should return 400 but returns 404
+		SkipConvey("When a root hierarchy node is requested with a invalid id", func() {
 
 			Convey("Then a 404 response code is returned", func() {
```
Comment (nshumoogum) [inline hierarchyAPI/get_root_hierarchy_test.go]: Again `400` instead of `404`

**Review by nshumoogum — CHANGES_REQUESTED** [review]

**Review by nshumoogum — APPROVED** [review]

**Pull request merged**